File: Clients/WindowsClient/WindowsClient/Messages/AuthenticationMessage.cs
using Newtonsoft.Json;
using Newtonsoft.Json.Linq;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace WindowsClient.Messages
{
    class AuthenticationMessage : Message
    {
        const String MESSAGE_TYPE = "Authentication";
        const String USERNAME_KEY = "Username";
        const String PASSWORD_KEY = "Password";
        const String AUTHENTICATED_KEY = "Authenticated";

        public String Username
        {
            get { return username; }
            set { username = value; }
        }
        public String Password
        {
            get { return password; }
            set { password = value; }
        }
        public Boolean Authenticated
        {
            get { return authenticated; }
            set { authenticated = value; }
        }

        String username;
        String password;
        Boolean authenticated;

        public AuthenticationMessage(String username, String password, Boolean authenticated)
        {
            this.username = username;
            this.password = password;
            this.authenticated = authenticated;
        }

        public AuthenticationMessage(JObject jobject)
        {
            JObject contents = JObject.Parse(jobject.GetValue(MESSAGE_TYPE).ToString());
            this.username = contents.GetValue(USERNAME_KEY).ToString();
            this.password = contents.GetValue(PASSWORD_KEY).ToString();
            this.authenticated = Convert.ToBoolean(contents.GetValue(AUTHENTICATED_KEY).ToString());
        }

        public JObject toJson()
        {
            JObject messageBuilder = new JObject();
            JObject thisBuilder = new JObject();
            thisBuilder.Add(USERNAME_KEY, this.username);
            thisBuilder.Add(PASSWORD_KEY, this.password);
            thisBuilder.Add(AUTHENTICATED_KEY, this.authenticated);

            messageBuilder.Add( MESSAGE_TYPE, thisBuilder);
            return messageBuilder;
        }

        public byte[] getMessageBytes()
        {
            ASCIIEncoding encoder = new ASCIIEncoding();
            return encoder.GetBytes( toJson().ToString(Formatting.None) );
        }

        public int getMessageLength()
        {
            return getMessageBytes().Length;
        }
    }
}

File: Server/src/main/java/Messages/AuthenticationMessage.java
package Messages;


import javax.json.Json;
import javax.json.JsonObject;
import javax.json.JsonObjectBuilder;

/**
 * Created by Saharath Kleips on 11/12/2015.
 */
public class AuthenticationMessage implements Message{
    static final String MESSAGE_TYPE = "Authentication";
    static final String USERNAME_KEY = "Username";
    static final String PASSWORD_KEY = "Password";
    static final String AUTHENTICATED_KEY = "Authenticated";

    public String getUsername() {
        return username;
    }

    public void setUsername(String username) {
        this.username = username;
    }

    public String getPassword() {
        return password;
    }

    public void setPassword(String password) {
        this.password = password;
    }

    public boolean isAuthenticated() {
        return authenticated;
    }

    public void setAuthenticated(boolean authenticated) {
        this.authenticated = authenticated;
    }

    String username;
    String password;
    boolean authenticated;

    public AuthenticationMessage(String username, String password, boolean authenticated) {
        this.username = username;
        this.password = password;
        this.authenticated = authenticated;
    }

    public AuthenticationMessage( JsonObject object )
    {
        JsonObject contents = JsonHelper.jsonFromString( object.get(MESSAGE_TYPE).toString() );
        this.username = contents.getString(USERNAME_KEY);
        this.password = contents.getString(PASSWORD_KEY);
        this.authenticated = Boolean.parseBoolean(contents.get(AUTHENTICATED_KEY).toString());
    }

    public JsonObject toJson() {
        JsonObjectBuilder messageBuilder = Json.createObjectBuilder();
        JsonObjectBuilder thisBuilder = Json.createObjectBuilder();
        thisBuilder.add(USERNAME_KEY, this.username)
                .add(PASSWORD_KEY, this.password)
                .add(AUTHENTICATED_KEY, this.authenticated);

        messageBuilder.add(this.MESSAGE_TYPE, thisBuilder);
        return messageBuilder.build();
    }

    public byte[] getMessageBytes() {
        return toJson().toString().getBytes();
    }

    public int getMessageLength() {
        return getMessageBytes().length;
    }
}
